notebooks/07 Model comparison.ipynb

```diff
@@ -0,0 +1,506 @@
+    "\n",
+    "Regression -> Length Of Stay -> Equivalent risk score"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "gather": {
+     "logged": 1653472153816
+    },
+    "jupyter": {
+     "outputs_hidden": false,
+     "source_hidden": false
+    },
+    "nteract": {
+     "transient": {
+      "deleting": false
+     }
+    }
+   },
+   "outputs": [],
+   "source": [
+    "# setup a subplot figure\n",
+    "fig, axs = plt.subplots(len(models_classification), 2)\n",
+    "fig.set_size_inches(15, 7 * len(models_classification))\n",
+    "\n",
+    "i = 0\n",
+    "\n",
+    "for model_classification in models_classification:\n",
+    "    if model_classification == \"catboost\":\n",
+    "        model_classification_X_test = X_test_catboost.drop(columns=\"risk\")\n",
+    "        model_classification_y_test = y_test\n",
+    "    elif model_classification == \"elastic\":\n",
+    "        model_classification_X_test = X_test_scaled.drop(columns=\"risk\")\n",
+    "        model_classification_y_test = y_test\n",
+    "    else:\n",
+    "        model_classification_X_test = X_test.drop(columns=\"risk\")\n",
+    "        model_classification_y_test = y_test\n",
+    "\n",
+    "    # elasticnet is scaled for classification (LogisticRegression), but not for regression (ElasticNet)\n",
+    "    if model_classification == \"elastic\":\n",
+    "        model_regression_X_test = X_test.drop(columns=\"risk\")\n",
+    "        model_regression_y_test = y_test\n",
+    "    else:\n",
+    "        model_regression_X_test = model_classification_X_test\n",
+    "        model_regression_y_test = model_classification_y_test\n",
+    "\n",
+    "    # prior is being compared to mean\n",
+    "    if model_classification == \"prior\":\n",
+    "        model_regression = \"mean\"\n",
+    "    else:\n",
+    "        model_regression = model_classification\n",
+    "\n",
+    "    # perform inference\n",
+    "    preds_regression = models_regression[model_regression][\"model\"].predict(\n",
+    "        model_regression_X_test\n",
+    "    )\n",
+    "    preds_classification = models_classification[model_classification][\"model\"].predict(\n",
+    "        model_classification_X_test\n",
+    "    )\n",
+    "\n",
+    "    # create a prediction dataframe\n",
+    "    predictions_df = pd.DataFrame(\n",
+    "        data=model_classification_y_test.reset_index(drop=True)\n",
+    "    )\n",
+    "    predictions_df[\"pred_regression_los\"] = preds_regression\n",
+    "    # calculate equivalent risk score from regression model\n",
+    "    predictions_df[\"pred_regression\"] = [\n",
+    "        risk_score(los) for los in predictions_df.pred_regression_los\n",
+    "    ]\n",
+    "    predictions_df[\"pred_classification\"] = preds_classification\n",
+    "\n",
+    "    #### Predicted vs Actual ####\n",
+    "\n",
+    "    # plot predicted vs actual CLASSES for classification\n",
```

Comment:
Not sure if it might be useful to include the metrics for each model above each plot too? Up to you if you think it'd add value, but could be useful to cross-reference what scores well vs. what looks good.